server/src/main/java/org/opensearch/index/snapshots/blobstore/BlobStoreRemStoreBasedIndexShardSnapshot.java

```diff
@@ -0,0 +1,263 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.snapshots.blobstore;
+
+import org.opensearch.OpenSearchParseException;
+import org.opensearch.core.ParseField;
+import org.opensearch.core.xcontent.ToXContentFragment;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.common.xcontent.XContentParserUtils;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Remote Store based Shard snapshot metadata
+ *
+ * @opensearch.internal
+ */
+public class BlobStoreRemStoreBasedIndexShardSnapshot implements ToXContentFragment {
+
+    private final String snapshot;
+    private final long indexVersion;
+    private final long startTime;
+    private final long time;
+    private final int incrementalFileCount;
+    private final long incrementalSize;
+    private final String primaryTermGenerationPair;
+
+    private final String remoteStoreRepository;
```

Comment:
Is storing the name good enough ? In future, one can register repository with same name but different base path. Wondering if we have to store some other information related to remote repository for validation ?

Reply:
we can maybe store repository base path as well here. however, even for remote store, we are just storing repository name in index settings.